src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const resObj: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const keyOfItem = item[key] as string;
```

Comment:
```suggestion
    const keyOfItem = String(item[key]);
```

Reply:
Thanks. i was looking for solution like this